Create a new react app (#23)
reactjs/pl.reactjs.org: content/docs/create-a-new-react-app.md

---
id: create-a-new-react-app
title: Stwórz nową aplikację w Reakcie
permalink: docs/create-a-new-react-app.html
redirect_from:
  - "docs/add-react-to-a-new-app.html"
prev: add-react-to-a-website.html
next: cdn-links.html
---

> Najwygodniej będzie ci się pracowało jeśli zastosujesz zintegrowany zestaw narzędzi (ang. toolchain). Użycie go pozwoli ci też zapewnić dobre doświadczenie użytkownikom twojej aplikacji.

Reviewer: ```suggestion
Najwygodniej będzie ci się pracowało, jeśli zastosujesz zintegrowany zestaw narzędzi (ang. *toolchain*). Użycie go pozwoli ci też zapewnić dobre doświadczenie użytkownikom twojej aplikacji.
```

Poniżej przedstawiamy kilka popularnych zestawów narzędziowych używanych w Reakcie. Pomagają one w zadaniach typu:

> * Zwiększanie skali projektu do wielu plików i komponentów.

Reviewer: Jestem za tym, żeby listy wypunktowane traktować jako jedno zdanie, więc zaczynać od małej litery, a kończyć przecinkiem lub średnikiem, natomiast listy numerowane traktować jak osobne zdania - zaczynać od wielkiej litery, a kończyć kropką.

* Stosowanie dodatkowych bibliotek dostępnych w npm.
* Wczesne wykrywanie często popełnianych błędów.

> * Ciągła, automatyczna edycja CSS i JS w trakcie tworzenia aplikacji.

Reviewer: Ta edycja raczej nie jest "automatyczna". Chodzi o to, że zmiany w plikach są na żywo odzwierciedlane na stronie, bez konieczności przeładowywania jej.

> * Optymalizacja kodu przed oddaniem aplikacji do użytku.

Reviewer: ```suggestion
* Optymalizacja zbudowanego kodu przed wdrożeniem na środowisko produkcyjne.
```

Author: Wdrażać _na_ czy wdrażać _do_?

Reviewer: Doczytałem trochę i wychodzi na to, że ani tak, ani tak :-) [Na stronie IBM](https://www.ibm.com/support/knowledgecenter/pl/SSYJ99_8.5.0/deploy/dep_ovr.html) piszą o "wdrożeniu kodu **w** środowisku produkcyjnym". Do tej pory myślałem, że to jest taka czynność... kierunkowa, ale to był, widać, skrót myślowy.

> Zestawy narzędziowe, które tutaj podajemy umożliwiają szybkie rozpoczęcie pracy and aplikacją i **nie wymagają żadnej wstępnej konfiguracji**

Reviewer: ```suggestion
Zestawy narzędziowe, które tutaj podajemy, umożliwiają szybkie rozpoczęcie pracy nad aplikacją i **nie wymagają żadnej wstępnej konfiguracji**.
```

## Stosowanie zestawu narzędziowego nie jest konieczne {#you-might-not-need-a-toolchain}

> Jeśli powyższe zadania nie przysparzają ci żadnych problemów lub jeśli nie wiesz jeszcze jak stosować narzędzia JavaScript, możesz [dodać React jako zwykły znacznik `<script>` do strony HTML](/docs/add-react-to-a-website.html), ewentualnie [z użyciem JSX](/docs/add-react-to-a-website.html#optional-try-react-with-jsx).

Reviewer: ```suggestion
Jeśli powyższe zadania nie przysparzają ci żadnych problemów lub jeśli nie wiesz jeszcze, jak stosować narzędzia JavaScript, możesz [dodać React jako zwykły znacznik `<script>` do strony HTML](/docs/add-react-to-a-website.html), ewentualnie [z użyciem JSX](/docs/add-react-to-a-website.html#optional-try-react-with-jsx).
```

> Jest to **najprostszy sposób zintegrowania Reacta z istniejącą już stroną internetową**. Zestaw narzędziowy możesz zawsze dodać do swojej aplikacji jeśli zajdzie taka potrzeba.

Reviewer: ```suggestion
Jest to **najprostszy sposób zintegrowania Reacta z istniejącą już stroną internetową**. Zestaw narzędziowy możesz zawsze dodać do swojej aplikacji, gdy pojawi się taka potrzeba.
```

## Zalecane zestawy narzędziowe {#recommended-toolchains}

Zespół Reacta rekomenduje następujące rozwiązania:

> - Jeśli **uczysz się Reacta** lub jeśli **tworzysz nową aplikację [jednostronicową](/docs/glossary.html#single-page-application),** skorzystaj z [Create React App](#create-react-app).

Reviewer: Nie podoba mi się określenie "jednostronicowa", bo raczej kojarzy mi się ze stronicą książki. Ale na tę chwilę nie mam lepszego tłumaczenia. Może ktoś inny będzie wiedział?

Author: `SPA` = `aplikacja jednostronicowa` pojawia się w kilku miejscach w sieci np:

https://docs.microsoft.com/pl-pl/aspnet/web-api/overview/getting-started-with-aspnet-web-api/build-a-single-page-application-spa-with-aspnet-web-api-and-angularjs

https://helion.pl/ksiazki/single-page-web-applications-programowanie-aplikacji-internetowych-z-javascript-michael-mikowski-josh-powell,sipawe.htm#format/d

Reviewer: Ok, widać nie czytam na co dzień literatury polskiej na ten temat, więc nie jestem przyzwyczajony do nomenklatury. Masz rację, trzeba dodać do Glosariusza na Slacku.

- Jeśli tworzysz **stronę internetową renderowaną na serwerze z użyciem Node.js,** spróbuj zastosować [Next.js](#nextjs).
- Jeśli tworzysz **stronę statyczną, która ma głównie wyświetlać określone treści,** spróbuj zastosować [Gatsby](#gatsby).

> - Jeśli tworzysz **bibliotekę kompnentów** lub **integrujesz z istniejącą bazą kodu,** spróbuj zastosować [bardziej elastyczne zestwy narzędziowe](#more-flexible-toolchains).

Reviewer: ```suggestion
- Jeśli tworzysz **bibliotekę komponentów** lub **integrujesz się z istniejącą bazą kodu**, spróbuj zastosować [bardziej elastyczne zestawy narzędziowe](#more-flexible-toolchains).
```

Zauważyłem, że najczęściej popełniasz literówki w słowie "komponent" :-)

Author: Może:

```suggestion
- Jeśli tworzysz **bibliotekę kompnentów** lub **dokonujesz integracji z istniejącą bazą kodu,** spróbuj zastosować [bardziej elastyczne zestwy narzędziowe](#more-flexible-toolchains).
```

Reviewer: Ok, faktycznie. Napisanie do czytelnika, że "integruje się z czymś" pasuje jedynie do "integrowania się z grupą" lub (w filmach sci-fi) integruje się z jakimś większym systemem, matrixem :-)
"Dokonujesz integracji" ma tu większy sens. Ale nadal (w Twojej sugestii) jest literówka w słowie "kompnentów" :-)

### Create React App {#create-react-app}

> [Create React App](http://github.com/facebookincubator/create-react-app) zapewnia bardzo dogodne środowisko pracy sprzyjające **nauce Reacta**. Jest to najlepszy sposób, aby zacząć tworzyć **nową [jednostronicową](/docs/glossary.html#single-page-application) aplikację** w Reakcie.

Reviewer: Ponownie, nie podoba mi się określenie "jednostronicowa".

> Środowisko pracy stworzone przez Create React App nie tylko umożliwi ci stosowanie najnowszych funkcjonalności JavaScript, lecz także zoptymalizuje twój kod przed oddaniem go do użytku i ogólnie znacznie usprawni twoją pracę. Aby móc skorzystać z tego rozwiązania na swoim komputerze, będziesz potrzebował Node >=6 oraz npm >=5.2. W celu stworzenia projektu zastosuj następujące komendy:

Reviewer: ```suggestion
Środowisko pracy stworzone przez Create React App nie tylko umożliwi ci stosowanie najnowszych funkcjonalności języka JavaScript, lecz także zoptymalizuje twój kod przed oddaniem go do użytku i ogólnie znacznie usprawni twoją pracę. Aby móc skorzystać z tego rozwiązania na swoim komputerze, będziesz potrzebować Node >= 6 oraz npm >= 5.2. Aby utworzyć projekt, wykonaj kolejno komendy:
```

```bash

> npx create-react-app my-app

Reviewer: ```suggestion
npx create-react-app moja-aplikacja
```
Tłumaczenie nazwy aplikacji, żeby podkreślić, iż jest to wartość dowolna.

> cd my-app

Reviewer: ```suggestion
cd moja-aplikacja
```

npm start
```

>Wskazówka

> > `npx` w pierwszej linijce powyżej to nie błąd w druku, to [narzędzie uruchamiające pakiety zawarte w npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).

Reviewer: ```suggestion
> `npx` w pierwszej linijce powyżej to nie literówka, to [narzędzie uruchamiające pakiety zawarte w npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).
```
W druku? W Internecie? :-)

Author: :-)

> Create React App nie obsługuje ani logiki backendu ani baz danych; tworzy jedynie front-endowy potok budowania (ang. *build pipeline*). Dzięki temu możesz go używać z dowolnie wybranym przez siebie backendem. Create React App zawiera [Babel](http://babeljs.io/) i [webpack](https://webpack.js.org/), ale nie musisz nic o nich wiedzieć, aby z powodzeniem używać tego zestawu narzędzi.

Reviewer: ```suggestion
Create React App nie obsługuje ani logiki backendu, ani baz danych; tworzy jedynie frontendowy potok budowania (ang. *build pipeline*). Dzięki temu możesz go używać z dowolnie wybranym przez siebie backendem. Create React App zawiera narzędzia takie, jak [Babel](http://babeljs.io/) i [webpack](https://webpack.js.org/), ale nie musisz nic o nich wiedzieć, aby z powodzeniem używać ich w projekcie.
```

> Kiedy uznasz, że twoja aplikacja jest gotowa do oddania do użytku, zastosuj komendę `npm run build`. Dzięki temu uzyskasz zoptymalizowaną wersję swojej aplikacji. Znajdziesz ją w folderze `build`. Więcej informacji na temat Create React App znajdziesz w [pliku README](https://github.com/facebookincubator/create-react-app#create-react-app-) oraz w tym [Przewodniku użytkownika](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents).

Reviewer: ```suggestion
Kiedy uznasz, że twoja aplikacja jest gotowa wdrożenia na środowisko produkcyjne, zastosuj komendę `npm run build`. Dzięki temu uzyskasz optymalnie zbudowaną wersję swojej aplikacji. Znajdziesz ją w folderze `build`. Więcej informacji na temat Create React App znajdziesz w [pliku README](https://github.com/facebookincubator/create-react-app#create-react-app-) oraz w tym [Przewodniku użytkownika](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents).
```

Author: jw.
wdrażać "na" czy "do"?

Reviewer: "na"

### Next.js {#nextjs}

> [Next.js](https://nextjs.org/) jest popularnym, lekkim frameworkiem służącym do budowy aplikacji reactowych **statycznych oraz renderowanych po stronie serwera**. Zestaw ten zawiera **rozwiązania stylizujące  i mechanizmy routingu**. Ponadto, Next.js domyślnie zakłada, że stosujesz [Node.js](https://nodejs.org/) jako środowisko serwera.

Reviewer: ```suggestion
[Next.js](https://nextjs.org/) jest popularnym, lekkim frameworkiem służącym do budowy aplikacji reactowych **statycznych oraz renderowanych po stronie serwera**. Zestaw ten zawiera **rozwiązania do tworzenia styli oraz mechanizmy routingu**. Ponadto, Next.js domyślnie zakłada, że stosujesz [Node.js](https://nodejs.org/) jako środowisko serwerowe.
```

Author: Z jakiegoś powodu liczba mnoga w "tworzeniu styli" nie gra mi. Może:

Zestaw ten zawiera **rozwiązania ułatwiające dodawanie stylu  i mechanizmy routingu**?

Reviewer: Ok, tak też jest dobrze, a przy okazji unikamy tego "zgrzytu".

Zapoznaj się z zestawem narzędziowym Next.js poprzez [oficjalny przewodnik](https://nextjs.org/learn/).

### Gatsby {#gatsby}

> [Gatsby](https://www.gatsbyjs.org/) zapewnia najlepszy sposób tworzenia **stron statycznych** z Reactem. Ten zestaw narzędziowy umożliwia korzystanie z komponentów reactowych, ale w wyniku jego zastosowania otrzymujemy wstępne renderowanie HTML i CSS co zapewnia najszczybsze możliwe ładowanie strony.

Reviewer: ```suggestion
[Gatsby](https://www.gatsbyjs.org/) to najlepsze narzędzie do tworzenia **statycznych stron** korzystających z Reacta. Pozwala na tworzenie struktury aplikacji przy użyciu komponentów reactowych, lecz jako wynik końcowy generuje wstępnie wyrenderowane pliki HTML i CSS w celu zapewnienia maksymalnej szybkości ładowania strony.
```

Zapoznaj się z Gatsbym poprzez [oficjalny przewodnik](https://www.gatsbyjs.org/docs/) oraz [galerię zestawów startowych](https://www.gatsbyjs.org/docs/gatsby-starters/).

### Bardziej elastyczne zestawy narzędziowe {#more-flexible-toolchains}

Podane niżej zestawy narzędziowe są stosunkowo bardziej elastyczne i dają większą dowolność konfiguracji. Polecamy je bardziej doświadczonym użytkownikom.

> - **[Neutrino](https://neutrinojs.org/)** łączy w sobie siłę [webpacku](https://webpack.js.org/) z prostotą ustawień domyślnych. Zestaw ten zawiera ustawienia domyślne dla [aplikacji reactowych](https://neutrinojs.org/packages/react/) i [komponentów reactowych](https://neutrinojs.org/packages/react-components/).

Reviewer: ```suggestion
- **[Neutrino](https://neutrinojs.org/)** łączy w sobie siłę [webpacka](https://webpack.js.org/) z prostotą ustawień domyślnych. Zestaw ten zawiera ustawienia domyślne dla [aplikacji reactowych](https://neutrinojs.org/packages/react/) i [komponentów reactowych](https://neutrinojs.org/packages/react-components/).
```

> - **[nwb](https://github.com/insin/nwb)** jest zalecany szczególnie dla potrzeb [publikowania komponentów reactowych na npm](https://github.com/insin/nwb/blob/master/docs/guides/ReactComponents.md#developing-react-components-and-libraries-with-nwb). Może też [być stosowany](https://github.com/insin/nwb/blob/master/docs/guides/ReactApps.md#developing-react-apps-with-nwb) do tworzenia aplikacji w Reakcie.

Reviewer: ```suggestion
- **[nwb](https://github.com/insin/nwb)** jest zalecany szczególnie do [publikowania komponentów reactowych na npm](https://github.com/insin/nwb/blob/master/docs/guides/ReactComponents.md#developing-react-components-and-libraries-with-nwb). Może też [być stosowany](https://github.com/insin/nwb/blob/master/docs/guides/ReactApps.md#developing-react-apps-with-nwb) do tworzenia aplikacji w Reakcie.
```

> - **[Parcel](https://parceljs.org/)** jest szybkim bundlerem aplikacji webowych, niewymagającym żadnego konfigurowania dla potrzeb [aplikacji reactowych](https://parceljs.org/recipes.html#react).

Reviewer: ```suggestion
- **[Parcel](https://parceljs.org/)** jest szybkim bundlerem aplikacji webowych, [doskonale współgrającym z Reactem](https://parceljs.org/recipes.html#react).
```

Author: Tak

> - **[Razzle](https://github.com/jaredpalmer/razzle)** jest frameworkiem umożliwiającym renderowanie po stronie serwera niewymagającym konfigurowania, ale dającym większy wybór niż Next.js.

Reviewer: ```suggestion
- **[Razzle](https://github.com/jaredpalmer/razzle)** jest frameworkiem umożliwiającym renderowanie po stronie serwera bez zbędnej konfiguracji, jednak dającym więcej możliwości niż Next.js.
```

## Tworzenie zestawu narzędziowego od podstaw {#creating-a-toolchain-from-scratch}

> W skład zestaw narzędziowy w JavaScripcie wchodzą zazwyczaj następujące elementy:

Reviewer: ```suggestion
W skład javascriptowych narzędzi do budowania wchodzą zazwyczaj następujące elementy:
```

Author: ```suggestion
W skład javascriptowych narzędzi do budowania aplikacji wchodzą zazwyczaj następujące elementy:
```

* **Menadżer pakietów** np. [Yarn](https://yarnpkg.com/) lub [npm](https://www.npmjs.com/). Umożliwia on korzystanie z ogromnego ekosystemu dodatkowych pakietów. Pozwala łatwo je instalować i aktualizować.

* **Bundler** np. [webpack](https://webpack.js.org/) lub [Parcel](https://parceljs.org/). Umożliwia on pisanie kodu modularnego i pakowania go w małe pakiety, aby zoptymalizować czas ładowania.

* **Kompilator** np. [Babel](http://babeljs.io/). Pozwala on na stosowanie nowych wersji JavaScriptu przy zachowaniu kompatybilności ze starszymi przeglądarkami.

> Jeśli chcesz stworzyć swój własny zestaw narzędziowy od podstaw, zajrzyj do [tego przewodnika](https://blog.usejournal.com/creating-a-react-app-from-scratch-f3c693b84658). Dowiesz się z niego jak odtworzyć niektóre z funkcjonalności Create React App.

Reviewer: ```suggestion
Jeśli chcesz stworzyć swój własny zestaw narzędziowy od podstaw, zajrzyj do [tego przewodnika](https://blog.usejournal.com/creating-a-react-app-from-scratch-f3c693b84658). Dowiesz się z niego, jak odtworzyć niektóre z funkcjonalności Create React App.
```

> Pamiętaj aby upewnić się, że twój własny zestaw narzędziowy [tworzy właściwą użytkową wersję aplikacji](/docs/optimizing-performance.html#use-the-production-build).

Reviewer: ```suggestion
Pamiętaj, aby upewnić się, że twój własny zestaw narzędziowy [jest przystosowany do wdrożeń produkcyjnych](/docs/optimizing-performance.html#use-the-production-build).
```

Author: tak
